bus-mapping/src/exec_trace.rs

```diff
@@ -0,0 +1,213 @@
+pub(crate) mod exec_step;
+use crate::evm::EvmWord;
+use crate::operation::{container::OperationContainer, Operation};
+use core::ops::{Index, IndexMut};
+pub(crate) use exec_step::ExecutionStep;
+use pasta_curves::arithmetic::FieldExt;
+
+/// Doc
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct BlockConstants<F: FieldExt> {
+    hash: EvmWord, // Until we know how to deal with it
+    coinbase: F,
+    timestamp: F,
+    number: F,
+    difficulty: F,
+    gaslimit: F,
+    chain_id: F,
+}
+
+/// Doc
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct ExecutionTrace<F: FieldExt> {
+    entries: Vec<ExecutionStep>,
+    block_ctants: BlockConstants<F>,
+    container: OperationContainer,
+}
+
+impl<F: FieldExt> Index<usize> for ExecutionTrace<F> {
+    type Output = ExecutionStep;
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.entries[index]
+    }
+}
+
+impl<F: FieldExt> IndexMut<usize> for ExecutionTrace<F> {
+    fn index_mut(&mut self, index: usize) -> &mut Self::Output {
+        &mut self.entries[index]
+    }
+}
+
+impl<F: FieldExt> ExecutionTrace<F> {
+    fn container_mut(&mut self) -> &mut OperationContainer {
+        &mut self.container
+    }
+
+    // Generates the trace setting the propper GC's and BusMappings for each step.
+    pub fn new(mut entries: Vec<ExecutionStep>, block_ctants: BlockConstants<F>) -> Self {
+        let mut container = OperationContainer::new();
+        let mut gc_counter = 0usize;
```

Comment:
nit: `gc_counter` would expands to `global_counter_counter` semantically, maybe just `gc` here?
```suggestion
        let mut gc = 0usize;
```